### PR TITLE
mem/file.go - Fix some races in accessing fields of FileData

### PR DESCRIPTION
**Pull request by corentone**
* Splitting SetModeTime to avoid double locking
* Adding locks all over the place.

Potential fix for #133

### CODE REVIEW

File: mem/file.go [hunk 1]
```diff
@@ -74,14 +74,24 @@ func CreateDir(name string) *FileData {
...
 
 func ChangeFileName(f *FileData, newname string) {
+	f.Lock()
+	defer f.Unlock()
 	f.name = newname
 }
 
 func SetMode(f *FileData, mode os.FileMode) {
+	f.Lock()
+	defer f.Unlock()
```
Comment (bep) [inline mem/file.go]: No need to use `defer` when the method is so short.

Comment (bep) [reply mem/file.go]: Also, it would be great if you added a test that demonstrates this problem.

**Comment by corentone** [comment]
@bep thanks for the review and sorry for the delay addressing them.
Let me know if you have any comment, I also added some UT for the funcs I changed which are kind of boringly verbose...

**Comment by mbertschler** [comment]
I verified that the new tests fail with data races without this patch.

**Comment by mbertschler** [comment]
Fixes #133

**Pull request merged**